### ISSUE TITLE
4.23.0 features are not available

### ISSUE THREAD

**Issue by usmanat77sparx** [adjust/corona_sdk#18]
It's a request to update CoronaSDK plugin to iOS/Android 4.23.0

**Comment by usmanat77sparx** [comment]
Hello Adjust team,

Do you guys have any plan to update the Solar2D plugin? Or possible to share iOS .a file and Android .jar file for 4.23.0 and 4.24.0 so we can pack them and use them as a plugin?

**Comment by ugi** [comment]
Hey @usmanat77sparx 

We were actually planning to update Corona SDK to v4.23.0. In case you're already using Solar2D platform, do you maybe know if Cordova plugin in way how we're releasing it is possible to be used in Solar2D app or does it need some adjustments?

**Comment by usmanat77sparx** [comment]
I don't think the Cordova plugin would be used. You can generate libplugin_adjust.a and plugin.adjust.jar just like you are doing in release page in the past.

**Comment by ugi** [comment]
Hey @usmanat77sparx 

We just added [v4.23.0](https://github.com/adjust/corona_sdk/releases/tag/v4.23.0) features to Corona plugin. Feel free to comment / reopen if you encounter any issue or have any further questions.

Cheers

**Comment by usmanat77sparx** [comment]
Hi @ugi, it seems the new Solar2D adjust plugin is not working. Here is a sample project.  [Adjust_4.23.0_Test_Project.zip](https://github.com/adjust/corona_sdk/files/5194340/Adjust_4.23.0_Test_Project.zip)

**Comment by ugi** [comment]
Hi @usmanat77sparx 

Which Solar2D version are you using? Is it [the latest one](https://github.com/coronalabs/corona/releases/tag/3616)?

**Comment by usmanat77sparx** [comment]
Hello @ugi,

Yes, I use the latest one 2020.3616 build. It says "duplicate symbols for architecture armv7". Here is the console error, it may help to fix the issue:

```
Analyzing Lua plugins...
Examining plugin: plugin.adjust
	Done.

Compiling Slices...
	Compiling architecture:	armv7
	<execute>cd "/var/folders/cy/vvzc2lx118bbnsv2y1sp_4lm0000gn/T/CLtmpLh6N1L/armv7";unset LIBRARY_PATH; export IPHONEOS_DEPLOYMENT_TARGET=8; export PATH="/Applications/Xcode11.6.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/:/Applications/Xcode11.6.app/Contents/Developer/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin";"/Applications/Xcode11.6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++" -stdlib=libstdc++ -arch armv7 -isysroot "/Applications/Xcode11.6.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS13.6.sdk" -ObjC -all_load -fobjc-link-runtime -miphoneos-version-min=8 -std=c++11 -stdlib=libc++ -Xlinker -rpath -Xlinker /usr/lib/swift -Xlinker -rpath -Xlinker @executable_path/Frameworks -L"/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate" -L"/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust" -F"/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate" -F"/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust" -framework AudioToolbox -framework AVFoundation -framework CFNetwork -framework CoreGraphics -framework CoreLocation -framework CoreMedia -framework CoreMotion -framework CoreVideo -framework CoreText -framework GameKit -framework GLKit -framework ImageIO -framework MapKit -framework MediaPlayer -framework MessageUI -framework MobileCoreServices -framework OpenAL -framework QuartzCore -framework Security -framework StoreKit -framework SystemConfiguration -framework UIKit -weak_framework GameController -weak_framework AssetsLibrary -weak_framework Photos -weak_framework Foundation -weak_framework Twitter -weak_framework AdSupport -weak_framework iAd -lobjc -lsqlite3 -ltemplate -lc++ -lz -lplugin_adjust -o "/var/folders/cy/vvzc2lx118bbnsv2y1sp_4lm0000gn/T/CLtmpLh6N1L/armv7/Adjust_4.23.0_Test_Project"</execute>
	<result>ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaProvider.gameNetwork.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(gameNetwork.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaGameNetworkLibrary.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaProvider.licensing.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(licensing.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaLicensingLibrary.o)) built for free standing
duplicate symbol '_OBJC_CLASS_$_ADJUrlStrategy' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_METACLASS_$_ADJUrlStrategy' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._gdprUrlChoicesArray' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._subscriptionUrlChoicesArray' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._baseUrlChoicesArray' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._choiceIndex' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._startingChoiceIndex' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._wasLastAttemptSuccess' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._overridenGdprUrl' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._overridenSubscriptionUrl' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._overridenBaseUrl' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._extraPath' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
ld: 12 duplicate symbols for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)</result>
ERROR: Builder failed: ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaProvider.gameNetwork.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(gameNetwork.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaGameNetworkLibrary.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaProvider.licensing.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(licensing.o)) built for free standing
ld: warning: building for iOS, but linking in object file (/Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/libtemplate/libtemplate.a(CoronaLicensingLibrary.o)) built for free standing
duplicate symbol '_OBJC_CLASS_$_ADJUrlStrategy' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_METACLASS_$_ADJUrlStrategy' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._gdprUrlChoicesArray' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._subscriptionUrlChoicesArray' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._baseUrlChoicesArray' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._choiceIndex' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._startingChoiceIndex' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._wasLastAttemptSuccess' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._overridenGdprUrl' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._overridenSubscriptionUrl' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._overridenBaseUrl' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
duplicate symbol '_OBJC_IVAR_$_ADJUrlStrategy._extraPath' in:
    /Users/usmanmughal/Desktop/Build/Adjust_4.23.0_Test_Project.app/.build/plugin.adjust/libplugin_adjust.a(ADJUrlStrategy-8702A38D30AAB0B9.o)
ld: 12 duplicate symbols for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)
BUILD ERROR: There was a problem linking the app.

Check the console for more information.
iOS build failed (2) after 6 seconds
ERROR: Build Failed: There was a problem linking the app.

Check the console for more information.
```

**Comment by ugi** [comment]
Hey @usmanat77sparx 

I think we have identified the issue. We're gonna release patch update for this soon and should hopefully solve it. Will keep you posted.

**Comment by ugi** [comment]
Hi @usmanat77sparx 

This issue should be addressed in our latest [v4.23.1](https://github.com/adjust/corona_sdk/releases/tag/v4.23.1) release. Feel free to give it a shot and reopen/comment in case issue is not fixed.

Cheers
